app/src/main/java/ch/epfl/sdp/blindly/match/MatchActivity.kt

```diff
@@ -0,0 +1,149 @@
+package ch.epfl.sdp.blindly.match
+
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import android.view.animation.AccelerateInterpolator
+import android.view.animation.DecelerateInterpolator
+import android.view.animation.LinearInterpolator
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.view.GravityCompat
+import androidx.drawerlayout.widget.DrawerLayout
+import androidx.recyclerview.widget.DefaultItemAnimator
+import androidx.recyclerview.widget.DiffUtil
+import com.yuyakaido.android.cardstackview.*
+import java.util.*
+import ch.epfl.sdp.blindly.R
+
+class MatchActivity : AppCompatActivity(), CardStackListener {
+
+    private lateinit var profiles : ArrayList<Profile>
+    private val drawerLayout by lazy { findViewById<DrawerLayout>(R.id.drawer_layout) }
+    private val cardStackView by lazy { findViewById<CardStackView>(R.id.card_stack_view) }
+    private val manager by lazy { CardStackLayoutManager(this, this) }
+    private val adapter by lazy { CardStackAdapter(profiles) }
+
+    /**
+     * Create the activity and setup the cardStack
+     *
+     * @param savedInstanceState
+     */
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        profiles = intent.getParcelableArrayListExtra("EXTRA_MATCH_PROFILES")!!
+        setContentView(R.layout.activity_match)
+        setupCardStackView()
+        setupButton()
+    }
+
+    override fun onBackPressed() {
+        if (drawerLayout.isDrawerOpen(GravityCompat.START)) {
+            drawerLayout.closeDrawers()
+        } else {
+            super.onBackPressed()
+        }
+    }
```

Comment:
Consider adding some documentation for this function, as I don't understand well what it does.